cli/src/bin.ts

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env node
+
+import { Command } from "commander";
+import * as path from "path";
+import fg from "fast-glob";
+import { watch } from "chokidar";
+import * as fs from "fs/promises";
+import { parseMachinesFromFile } from "xstate-parser-demo";
+import {
+  doesTsTypesRequireUpdate,
+  FileEdit,
+  makeXStateUpdateEvent,
+  processFileEdits,
+  writeToTypegenFile,
+} from "xstate-tools-shared";
+
+const program = new Command();
+
+program.version("0.0.1");
+
+const handleError = (uri: string, e: any) => {
+  if (e?.code === "BABEL_PARSER_SYNTAX_ERROR") {
+    console.log(`${uri} - syntax error, skipping`);
+  }
+};
+
+const writeToFiles = async (uriArray: string[]) => {
+  /**
+   * TODO - implement batching to speed things up
```

Comment:
This is just about wrapping a single iteration of the loop in a promise and calling Promise.all on the created promises, right? seems like an easy change to be included as part of this PR

Reply:
Iiish, we don't want to run 40+ checks at the same time. Although maybe that's better than not having it.

Reply:
Ye, not feeling super strong about it - this could always be chunked/backpressured if we see that too much parallelization is not that good. I'm happy with the current waterfall solution for now though. I've just thought that maybe you want to act on this comment soon(ish)